.github/workflows/scripts/e2e/gmc_xeon_test.sh

```diff
@@ -452,10 +452,14 @@ function validate_modify_config() {
     #change the model id of the step named "Tgi" in the codegen_xeon_mod.yaml
     yq -i '(.spec.nodes.root.steps[] | select ( .name == "Tgi")).internalService.config.MODEL_ID = "HuggingFaceH4/mistral-7b-grok"' $(pwd)/config/samples/CodeGen/codegen_xeon_mod.yaml
     kubectl apply -f $(pwd)/config/samples/CodeGen/codegen_xeon_mod.yaml
-    #you are supposed to see an error, it's a known issue, but it does not affect the tests
-    #https://github.com/opea-project/GenAIInfra/issues/314
 
-    pods_count=$(kubectl get pods -n $MODIFY_STEP_NAMESPACE -o jsonpath='{.items[*].metadata.name}' | wc -w)
+    # Wait until all pods are ready
+    wait_until_all_pod_ready $MODIFY_STEP_NAMESPACE 300s
```

Comment:
This could leads test failure. `wait_until_all_pod_ready` doesn't support the case when some pods are being deleted.

Reply:
thanks for reminding, I will keep monitoring it.

Reply:
Could you please enhance `wait_until_all_pod_ready` so it can omit those pods with `Terminating` status when it tries to get the list of pods to wait for ready?

Reply:
yes, I am just about to do it